### PR TITLE
Improve "first contact" aspect in README file

### PR DESCRIPTION
**Pull request by owickstrom**
Hi,

As discussed in person at ZuriHac, but based on #55, we've fleshed out the README to include an overview of the library. There are more things in that issue that we can post PRs for later, but this felt like a decent start.

We have added:

* Synopsis/header
* Structure
  * Lifting
  * Transformers
* Resources

Big thanks to @gilligan for helping me out with this PR!

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -1,4 +1,145 @@
...
+    without any modification.
+
+    - Transformer: `Control.Monad.Trans.Identity.IdentityT` (in the `transformers` package)
+    - Identity functor and monad: `Data.Functor.Identity.Identity` (in the `base` package)
+
+* `Control.Monad.List`
+
+    This transformer combines the list monad with another monad. _Note
+    that this does not yield a monad unless the inner monad is
+    [commutative](https://en.wikipedia.org/wiki/Commutative_property)._
```
Comment (ocramz) [inline README.markdown]: What would be a good definition of a commutative monad? Something for which `m1 >>= m2 >>= f == m2 >>= m1 >>= k` ?

File: README.markdown [hunk 2]
```diff
@@ -1,4 +1,145 @@
...
+    A convenient transformer that combines the Reader, Writer, and
+    State monad transformers.
+
+    - Lazy transformer: `Control.Monad.RWS.Lazy.RWST` (which is the default, exported by `Control.Monad.RWS`)
+    - Strict transformer: `Control.Monad.RWS.Strict.RWST`
+
+* `Control.Monad.Reader`
+
+    The Reader monad transformer represents a computation which can
+    read values from a shared environment.
```
Comment (ocramz) [inline README.markdown]: I think "shared" is misleading here

File: README.markdown [hunk 3]
```diff
@@ -1,4 +1,145 @@
...
+    - Class: `Control.Monad.State.Class.MonadState`
+    - Lazy transformer: `Control.Monad.State.Lazy.StateT` (the default, exported by `Control.Monad.State`)
+    - Strict transformer: `Control.Monad.State.Strict.StateT`
+
+* `Control.Monad.Writer`
+
+    The Writer monad transformer represents a computation that can
+    produce a stream of data in addition to the computed values. This
+    can be used to collect values in some data structure with a
+    `Monoid` instance.
```
Comment (ocramz) [inline README.markdown]: .. for example for logging purposes

File: README.markdown (hunk 1 above)
Comment (owickstrom) [reply README.markdown]: `k` should be `f` in the end, right? Do you think it should be added, or is the link (or another link) good enough?

File: README.markdown (hunk 2 above)
Comment (owickstrom) [reply README.markdown]: Agreed, I'll change it!

File: README.markdown (hunk 3 above)
Comment (owickstrom) [reply README.markdown]: :+1: adding

File: README.markdown (hunk 1 above)
Comment (ocramz) [reply README.markdown]: Fixed thx :) Maybe having the example here can help someone visualize better

**Comment by owickstrom** [comment]
Thanks @ekmett !

**Pull request merged**